doc/classes/Vector3.xml

```diff
@@ -252,6 +252,13 @@
 				Returns a new vector moved toward [param to] by the fixed [param delta] amount. Will not go past the final value.
 			</description>
 		</method>
+		<method name="normalize_and_get_length">
+			<return type="float" />
+			<description>
+				Scales the vector to unit length, and returns the original length.
+				If the original vector was too small to normalize, the length returned is [code]zero[/code].
```

Comment:
There are also some issues with it because of access issues with GDScript, primitive types are non-mutable when accessed as properties, unless that has changed

Reply:
If you can think of an easy way to return two values efficiently from bound functions, I'm happy to modify the PR.

If we don't want it to mutate the struct, we need to return the new Vector _and_ the length (or the new Vector _and_ success or failure).

Originally this was non-bound, but it is now also a potential candidate solution to the problem in #74852, and whichever solution we go with needs to be available from script.

Another non-mutating alternative would be a `normalized()` function that returns either the normalized vector or failure (maybe a Vector or NULL? can that work?).

Reply:
Perhaps then for binding returning a variant that is either the normalized value or NULL is the way to go `try_normalized()` or something like that?

I can modify this particular PR to have this function unbound as before, but will be good to get some discussion going on how to solve the gdscript problem.

Reply:
It could return an array or dictionary, somewhat cumbersome and inefficient but it gets it done